### PR TITLE
fix: modified tests for step 10 of building a spam filter

### LINKED ISSUE #54337: Building a Spam Filter (Step 10): Passes with incorrect values if right-hand side of `|` is empty
### Describe the Issue

The empty right-hand side of `|` makes it match on anything.

```
/whatever|/.test('not');
// true
```

Both tests will pass with incorrect values as long as the right-hand side of `|` is empty.

```
const helpRegex = /test|/i;
```

tests: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-regular-expressions-by-building-a-spam-filter/641cf198ec366c33d6504854.md

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-regular-expressions-by-building-a-spam-filter/step-10

### Your code

```



```

### Expected behavior

Proper value checks.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

Forum: https://forum.freecodecamp.org/t/bug-report-learning-regular-expressions-by-building-a-spam-filter-step-10/684561

### PR DESCRIPTION
**Pull request by jnahan**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #54337 

Added checks to prevent step 10 of building a spam filter from passing if left-hand or right-hand side of `|` is empty

Screen Shot 2024-04-11 at 5 07 25 PM

### CODE REVIEW

**Comment by Sembauke** [comment]
image
I am still able to pass it this way

**Comment by Ksound22** [comment]
> I am still able to pass it this way image

That didn't pass on my end
Screenshot 2024-04-15 at 10 16 55

**Comment by Sembauke** [comment]
> That didn't pass on my end

Yup @jenna5376 fixed it in the latest commit :)

**Review by Sembauke — APPROVED** [review]

**Comment by Ksound22** [comment]
@Sembauke what do we do about [#54378](https://github.com/freeCodeCamp/freeCodeCamp/pull/54378)?

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-regular-expressions-by-building-a-spam-filter/641cf198ec366c33d6504854.md [hunk 1]
```diff
@@ -23,12 +23,16 @@ Your `helpRegex` should match the string `please help`.
 
 ```js
 assert.match('please help', helpRegex);
+const splitRegex = helpRegex.toString().split(/[\/|]/);
+assert.isTrue(splitRegex.indexOf('please help') != -1);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-regular-expressions-by-building-a-spam-filter/641cf198ec366c33d6504854.md:27]:
```suggestion
assert.include(splitRegex, 'please help');
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-regular-expressions-by-building-a-spam-filter/641cf198ec366c33d6504854.md [hunk 2]
```diff
@@ -23,12 +23,16 @@ Your `helpRegex` should match the string `please help`.
...
 ```
 
 Your `helpRegex` should match the string `assist me`.
 
 ```js
 assert.match('assist me', helpRegex);
+const splitRegex = helpRegex.toString().split(/[\/|]/);
+assert.isTrue(splitRegex.indexOf('assist me') != -1);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-regular-expressions-by-building-a-spam-filter/641cf198ec366c33d6504854.md:35]:
```suggestion
assert.include(splitRegex, 'assist me');
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Let's use the more specific assert method.

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**